app/src/pages/CounterPage/containers/CounterPageContainer.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react'
+
+import Counter from '../components/index';
+
+class CounterPageContainer extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            countValue: 0,
+            parityType: 'even'
+            
+        }
+    }
+
+    handleIncrement = () => {
+        this.setState((state) => {
+            return {
+                countValue: state.countValue + 1
+            }
+        })
+    }
+
+    handleDecrement = () => {
+        this.setState((state) => {
+            if (state.countValue > 0) return {
+                countValue: state.countValue - 1
+            } 
+        
+        })
```

Comment:
setState method run every time. Try to rewrite your logic in the following manner: if -> run setState method else do nothing